File: packages/python-sdk/src/blindfold/regex/detectors/be/national_number.py
"""Belgian National Number (Rijksregisternummer) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import be_national_number_checksum


@register_region("be")
class NationalNumberDetector(RegexDetector):
    entity_type = EntityType.BE_NATIONAL_NUMBER
    score = 0.85
    context_keywords = [
        "rijksregisternummer", "national number", "numero national",
        "registre national", "nn",
    ]
    context_required = True
    context_window = 50

    pattern = re.compile(
        r"\b\d{2}[. ]?\d{2}[. ]?\d{2}[- ]?\d{3}[. ]?\d{2}\b"
    )
    validator = staticmethod(be_national_number_checksum)

File: packages/js-sdk/src/regex/detectors/be/national-number.ts
// Belgian National Number (Rijksregisternummer) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { beNationalNumberChecksum } from '../../validators'

class BeNationalNumberDetector extends RegexDetector {
  entityType = EntityType.BE_NATIONAL_NUMBER
  score = 0.85
  contextKeywords = [
    'rijksregisternummer',
    'national number',
    'numero national',
    'registre national',
    'nn',
  ]
  contextRequired = true
  contextWindow = 50

  pattern = /\b\d{2}[. ]?\d{2}[. ]?\d{2}[- ]?\d{3}[. ]?\d{2}\b/g
  validator = beNationalNumberChecksum
}

registerRegion('be', BeNationalNumberDetector)

export { BeNationalNumberDetector }
